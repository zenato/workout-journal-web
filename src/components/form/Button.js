import React from 'react';
import classNames from 'classnames/bind';
import styles from './Button.scss';

const cx = classNames.bind(styles);

const Button = ({ type, className, ...props }) => (
  <input {...props} type={type || 'button'} className={cx('btn', className)} />
);

export default Button;
