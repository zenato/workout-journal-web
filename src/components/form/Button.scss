@import 'common';

.btn {
  background-color: #fff;
  color: #323232;
  padding: 10px;
  border: solid #d4d4d4 1px;
  cursor: pointer;

  $color: #161616;
  &:hover {
    color: $color;
    background-color: #f6f6f6;
  }
  &:active {
    color: $color;
    background-color: #efefef;
  }

  &.primary {
    color: #fff;
    background-color: #008ee0;
    &:hover {
      background-color: #00a4ff;
    }
  }
}

