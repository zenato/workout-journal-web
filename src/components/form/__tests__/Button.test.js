import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button', () => {
  const props = {
    type: 'submit',
    value: 'test',
    onClick: jest.fn(),
  };

  it('should render passed props.', () => {
    const component = shallow(<Button {...props} />);
    expect(component).toMatchSnapshot();
  });
});
