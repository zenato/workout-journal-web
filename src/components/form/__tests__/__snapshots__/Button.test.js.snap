// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render passed props. 1`] = `
<input
  className="btn"
  onClick={[Function]}
  type="submit"
  value="test"
/>
`;
